File: algorithms/warmup/fib_sum_ld.cpp
#include <iostream>
#include <vector>

int fib_mod(int64_t n) {
    int pp = 60;
    int fib_i = n % pp;
    std::vector<int> mod_seq = {0, 1};

    for (int i = 1; i < fib_i; i++) {
        int size = mod_seq.size();
        int res = (mod_seq.back() + mod_seq[size - 2]) % 10;
        mod_seq.push_back(res);
    }

    int sum = 0;

    for (int i = 0; i <= fib_i; i++) {
        sum += mod_seq[i];
    }

    return sum % 10;
}

int main() {
    int64_t n;
    std::cin >> n;
    std::cout << fib_mod(n);

    return 0;
}

File: algorithms/warmup/fib_sum_ld.py
#python3                                                                      
# Given an integer n, find the last digit of the sum F 0 + F 1 + · · · + F n .
                                                                              
def fib_mod_sum(n):                                                           
    mod_seq = [0, 1]                                                          
    # mod is 10 -> pisano period is 60                                        
    pp = 60                                                                   
                                                                              
    fib_i = n % pp                                                            
                                                                              
    for _ in range(1, fib_i):                                                 
        mod_seq.append((mod_seq[-1] + mod_seq[-2]) % 10)                      
                                                                              
    return sum(mod_seq[:fib_i + 1]) % 10                                      
                                                                              
                                                                              
def main():                                                                   
    n = int(input())                                                          
    print(fib_mod_sum(n))                                                     
                                                                              
                                                                              
if __name__ == "__main__":                                                    
    main()
                                                                    
